packages/dev/core/src/Rendering/GaussianSplatting/gaussianSplatting.ts

```diff
@@ -90,23 +94,43 @@ export class GaussianSplatting {
     protected _worker: Nullable<Worker> = null;
     protected static _VertexShaderSource = `
         precision mediump float;
+
+        attribute float splatIndex;
+
         attribute vec2 position;
 
-        attribute vec4 world0;
-        attribute vec4 world1;
-        attribute vec4 world2;
-        attribute vec4 world3;
+        uniform highp sampler2D covariancesATexture;
+        uniform highp sampler2D covariancesBTexture;
+        uniform highp sampler2D centersTexture;
+        uniform highp sampler2D colorsTexture;
+        uniform vec2 dataTextureSize;
 
-        uniform mat4 projection, modelView;
+        uniform mat4 projection;
+        uniform mat4 modelView;
         uniform vec2 viewport;
 
         varying vec4 vColor;
         varying vec2 vPosition;
+
+        // webgl context does not offer function transpose
+        mat3 _transpose(mat3 matrix) {

```

Comment:
You should do something like this, to keep the fast path in anything but WebGL1:
```glsl
#if !defined(WEBGL2) && !defined(WEBGPU) && !defined(NATIVE)
        mat3 transpose(mat3 matrix) {
            return mat3(matrix[0][0], matrix[1][0], matrix[2][0],
                matrix[0][1], matrix[1][1], matrix[2][1],
                matrix[0][2], matrix[1][2], matrix[2][2]);
        }
#endif
```
and use `transpose` instead of `_transpose` at the calling sites.